src/main/java/com/softserveinc/dokazovi/service/impl/PostServiceImpl.java

```diff
@@ -126,15 +126,10 @@ public Page<PostDTO> findAllByTypesAndStatusAndDirectionsAndOriginsAndTitleAndAu
 			Sort sort = pageable.getSort().and(Sort.by("modified_at").descending());
 			pageable = PageRequest.of(pageable.getPageNumber(), pageable.getPageSize(), sort);
 		}
-
-		Optional<LocalDateTime> startDate1 = Optional.ofNullable(startDate);
-		LocalDateTime startTime = startDate1
-				.orElse(LocalDateTime.of(LocalDate.EPOCH, LocalTime.MIN));
-		Timestamp startDateTimestamp = Timestamp.valueOf(startTime);
-		Optional<LocalDateTime> endDate1 = Optional.ofNullable(endDate);
-		LocalDateTime endTime = endDate1
-				.orElse(LocalDateTime.of(LocalDate.now(), LocalTime.MAX));
-		Timestamp endDateTimestamp = Timestamp.valueOf(endTime);
+		LocalDate startLocalDate = Optional.ofNullable(startDate).orElse(LocalDate.EPOCH);
+		LocalDate endLocalDate = Optional.ofNullable(endDate).orElse(LocalDate.now());
+		Timestamp startDateTimestamp = Timestamp.valueOf(startLocalDate.atStartOfDay());
+		Timestamp endDateTimestamp = Timestamp.valueOf(endLocalDate.atStartOfDay());
```

Comment:
Will it work correctly if the publication date "today" but the time will be out of `endLocalDate.atStartOfDay()` time range?
For example, we are requesting posts from 2022.08.1 00:00:00 till today (2022.08.29). And the time of `endDateTimestamp` will be applied like 00:00:00 if I'm not mistaken. Then publication from 2022.08.29 15:24:00 won't be included. Is that right?

If so, can we apply something like `endLocalDate.atEndOfDay()` or add 1 day more so that the `endDateTimestamp` was pointing out to tomorrow start of the day (2022.08.30 00:00:00)?